recommenders/models/embedding_ranker/nn_embedding_ranker.py

```diff
@@ -0,0 +1,516 @@
+    
+    def __init__(
+        self,
+        n_factors=32,
+        hidden_units=[64, 32],
+        n_epochs=10,
+        batch_size=64,
+        learning_rate=0.001,
+        weight_decay=0.0,
+        num_workers=0,
+        use_cuda=True,
+        seed=None,
+        verbose=False,
+    ):
+        """Initialize model parameters.
+        
+        Args:
+            n_factors (int): Dimension of the embedding vectors.
+            hidden_units (list): List of hidden unit sizes for prediction layers.
+            n_epochs (int): Number of epochs for training.
+            batch_size (int): Number of examples per training batch.
+            learning_rate (float): Learning rate for the optimizer.
+            weight_decay (float): L2 regularization strength.
+            num_workers (int): Number of workers for data loading.
+            use_cuda (bool): Whether to use GPU if available.
+            seed (int): Random seed for reproducibility.
+            verbose (bool): Whether to show training progress.
+        """
+        self.n_factors = n_factors
+        self.hidden_units = hidden_units
+        self.n_epochs = n_epochs
+        self.batch_size = batch_size
+        self.learning_rate = learning_rate
+        self.weight_decay = weight_decay
+        self.num_workers = num_workers
+        self.use_cuda = use_cuda and torch.cuda.is_available()
+        self.verbose = verbose
+        self.model = None
+        
+        # Set random seeds for reproducibility
+        if seed is not None:
+            np.random.seed(seed)
+            torch.manual_seed(seed)
+            if self.use_cuda:
+                torch.cuda.manual_seed_all(seed)
+                
+        self.device = torch.device("cuda" if self.use_cuda else "cpu")
+        logger.info(f"Using device: {self.device}")
+        
+    def fit(self, train_df, col_user=DEFAULT_USER_COL, col_item=DEFAULT_ITEM_COL, 
+            col_rating=DEFAULT_RATING_COL, val_df=None):
+        """Train the embedding ranker model.
+        
+        Args:
+            train_df (pandas.DataFrame): Training data.
+            col_user (str): User column name.
+            col_item (str): Item column name.
+            col_rating (str): Rating column name.
+            val_df (pandas.DataFrame, optional): Validation data.
+            
+        Returns:
+            NNEmbeddingRanker: Trained model instance.
+        """
+        # Create mapping of user/item IDs to internal indices
+        self.user_id_map = {user_id: i for i, user_id in enumerate(train_df[col_user].unique())}
+        self.item_id_map = {item_id: i for i, item_id in enumerate(train_df[col_item].unique())}
+        self.n_users = len(self.user_id_map)
+        self.n_items = len(self.item_id_map)
+        
+        logger.info(f"Number of unique users: {self.n_users}")
+        logger.info(f"Number of unique items: {self.n_items}")
+        
+        # Create mapped dataframes
+        train_mapped_df = train_df.copy()
```

Comment:
do we really need this? if the data is large, this could create a large overhead?